src/pipeline/pipeline.h

```diff
@@ -36,6 +36,8 @@ namespace librealsense
             std::shared_ptr<device_interface> wait_for_device(const std::chrono::milliseconds& timeout = std::chrono::hours::max(),
                 const std::string& serial = "");
             std::shared_ptr<librealsense::context> get_context() const;
+            void set_device( std::shared_ptr< librealsense::device_interface >  device_interface );
```

Comment:
It is better not to use argument name that is also a type name. Some compilers will issue errors on this.

Reply:
Changed